pkg/localmetrics/localmetrics.go

```diff
@@ -105,3 +132,75 @@ func UpdateMetricPagerDutyHeartbeat(APIKey string, timer *prometheus.Timer) {
 	}
 	MetricPagerDutyHeartbeat.Observe(float64(0))
 }
+
+// AddAPICall observes metrics for a call to an external API
+// - param controller: The name of the controller making the API call
+// - param req: The HTTP Request structure
+// - param resp: The HTTP Response structure
+// - param duration: The number of seconds the call took.
+func AddAPICall(controller string, req *http.Request, resp *http.Response, duration float64) {
+	ApiCallDuration.With(prometheus.Labels{
+		"controller": controller,
+		"method":     req.Method,
+		"resource":   resourceFrom(req.URL),
+		"status":     resp.Status,
+	}).Observe(duration)
+}
+
+// resourceFrom normalizes an API request URL, including removing individual namespace and
+// resource names, to yield a string of the form:
+//     $group/$version/$kind[/{NAME}[/...]]
+// or
+//     $group/$version/namespaces/{NAMESPACE}/$kind[/{NAME}[/...]]
+// ...where $foo is variable, {FOO} is actually {FOO}, and [foo] is optional.
+// This is so we can use it as a dimension for the apiCallCount metric, without ending up
```

Comment:
nit:

```suggestion
// This is so we can use it as a dimension for the ApiCallDuration metric, without ending up
```

This probably means I messed this up in a-a-o, huh?

Reply:
most likely